akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowExpandSpec.scala

```diff
@@ -97,43 +91,41 @@ class FlowExpandSpec extends AkkaSpec {
     }
 
     "backpressure publisher when subscriber is slower" in {
-      val publisher = StreamTestKit.PublisherProbe[Int]()
-      val subscriber = StreamTestKit.SubscriberProbe[Int]()
-
-      Source(publisher).expand(seed = i ⇒ i)(extrapolate = i ⇒ (i, i)).runWith(Sink(subscriber))
+      val publisher = TestPublisher.auto[Int]()
+      val subscriber = TestSubscriber.probe[Int]()
 
-      val autoPublisher = new StreamTestKit.AutoPublisher(publisher)
+      Source(publisher.probe).expand(seed = i ⇒ i)(extrapolate = i ⇒ (i, i)).to(Sink(subscriber)).run()
       val sub = subscriber.expectSubscription()
 
-      autoPublisher.sendNext(1)
+      publisher.sendNext(1)
       sub.request(1)
       subscriber.expectNext(1)
       sub.request(1)
       subscriber.expectNext(1)
 
-      var pending = autoPublisher.pendingRequests
+      var pending = publisher.pending
       // Deplete pending requests coming from input buffer
       while (pending > 0) {
-        autoPublisher.subscription.sendNext(2)
+        publisher.sendNextDisregardDemand(2)
```

Comment:
I meant that it wraps everything that the autoPublisher has except the tracked demand. The autoPublisher has the subscription AFAIK, and then it can give it to the manual probe.